### PR TITLE
Cleaning up types

### PR DESCRIPTION
**Pull request by kcoleman731**
* Renamed calendar associated resources. 
* Added calendar group types and names. 
* Added `template` type to publication status
* Updated display settings properties.

### CODE REVIEW

File: dist/models/calendar.d.ts [hunk 1]
```diff
@@ -38,9 +39,13 @@ declare namespace Calendar {
         };
         iCal?: string;
     }
-    enum CalendarGroupName {
+    enum GroupName {
```
Comment (tayhalla) [inline dist/models/calendar.d.ts]: Thanks for doing this namespacing rework

File: dist/models/calendar.js [hunk 2]
```diff
@@ -2,17 +2,21 @@
...
-        CalendarGroupType["Social"] = "SOCIAL";
-        CalendarGroupType["Dining"] = "DINING";
-        CalendarGroupType["ServiceProvider"] = "SERVICE_PROVIDER";
-    })(CalendarGroupType = Calendar.CalendarGroupType || (Calendar.CalendarGroupType = {}));
-    var CalendarGroupName;
-    (function (CalendarGroupName) {
-        CalendarGroupName["Club"] = "Club";
-        CalendarGroupName["Golf"] = "Golf";
-    })(CalendarGroupName = Calendar.CalendarGroupName || (Calendar.CalendarGroupName = {}));
+    var GroupType;
```
Comment (tayhalla) [inline dist/models/calendar.js]: @kcoleman731 why is `dist` checked in?

File: dist/models/calendar.d.ts [hunk 3]
```diff
@@ -65,7 +70,7 @@ declare namespace Calendar {
         publicBookings: boolean;
         joinableBookings: boolean;
         hours: HoursOfOperation[];
-        privileges: Privilege[];
+        privileges?: Privilege[];
```
Comment (tayhalla) [inline dist/models/calendar.d.ts]: Do we want to do this? If it's a collection we have to check A) existence then B) length as opposed to just checking length

File: dist/models/calendar.js (hunk 2 above)
Comment (kcoleman731) [reply dist/models/calendar.js]: because this is used as a lib.

File: dist/models/calendar.d.ts (hunk 3 above)
Comment (kcoleman731) [reply dist/models/calendar.d.ts]: fixed

**Pull request merged**